packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1000,12 +1000,14 @@ class _Dial extends StatefulWidget {
     @required this.mode,
     @required this.use24HourDials,
     @required this.onChanged,
+    @required this.onHourSelected,
```

Comment:
Even though this is private etc, It looks like the implementation assumes that onHourSelected is non-null; should assert as much here. If there similar assumptions about any of the other parameters, maybe more asserts too.